src/shogun/machine/GLM.cpp

```diff
@@ -0,0 +1,170 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Author: Tej Sukhatme
+ */
+
+#include <shogun/features/DotFeatures.h>
+#include <shogun/labels/Labels.h>
+#include <shogun/labels/RegressionLabels.h>
+#include <shogun/lib/observers/ObservedValueTemplated.h>
+#include <shogun/machine/GLM.h>
+#include <shogun/machine/LinearMachine.h>
+#include <shogun/mathematics/NormalDistribution.h>
+#include <shogun/mathematics/RandomNamespace.h>
+#include <shogun/mathematics/linalg/LinalgNamespace.h>
+#include <shogun/optimization/ConstLearningRate.h>
+#include <shogun/optimization/ElasticNetPenalty.h>
+#include <shogun/optimization/GradientDescendUpdater.h>
+#include <shogun/optimization/SGDMinimizer.h>
+
+#include <cmath>
+
+using namespace shogun;
+
+GLM::GLM()
+{
+	SG_ADD(
+	    &distribution, "distribution_type",
+	    "variable to store name of distribution type",
+	    ParameterProperties::HYPER);
```

Comment:
this will allow users to specify the enum as a string when using put/get
```suggestion
	    ParameterProperties::HYPER, SG_OPTIONS(POISSON)));
```